app.tsx

```diff
@@ -57,13 +71,54 @@ const App = () => {
   const [emailError, setEmailError] = useState(false);
   const [password, setPassword] = useState('');
   const [passwordError, setPasswordError] = useState(false);
-
+  const [authError, setAuthError] = useState(false);
+  
+  const [token, setToken] = useState('');
   
+  const storeToken = async (value: string) => {
+    try {
+      await AsyncStorage.setItem('@token', value)
+    } catch (e) {
+      console.log(e);
+    }
+  }
+
+  const getToken = async () => {
+    try {
+      const value = await AsyncStorage.getItem('@token')
+      if(value !== null) {
+        setToken(value);
+      }
+      else{
+        setToken('');
+      }
+    } catch(e) {
+      console.log(e);
+    }
+  }
 
+  const [loginMutation, {loading, error}] = useMutation(loginGQL, 
+    {
+      client: client, 
+      onCompleted: (data) => {
+        storeToken(data.login.token);
+      },
+      onError: () => {
+        setAuthError(true);
+      }
+    });
+  
   const handleSubmit = () => {
-    // Validating Email and Password
-    setEmailError(!emailValidator.test(email));
-    setPasswordError(!passwordValidator.test(password))
+    const isValidEmail = !emailValidator.test(email);
```

Comment:
Dado que vc "inverte" a booleana duas vezes, não faz mais sentido assim?
```suggestion
    const isValidEmail = emailValidator.test(email);
    const isValidPassword = passwordValidator.test(password);
    
    if(isValidEmail && isValidPassword) { ... }
```